include/seqan3/range/view/repeat.hpp

```diff
@@ -0,0 +1,325 @@
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides the view::repeat_view.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <range/v3/view/single.hpp>
+
+#include <seqan3/core/metafunction/iterator.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3::detail
+{
+
+// ---------------------------------------------------------------------------------------------------------------------
+// repeat_view class
+// ---------------------------------------------------------------------------------------------------------------------
+
+/*!\brief The type returned by seqan3::view::repeat.
+ * \tparam value_t The type of value to repeat which is always wrapped in a std::view::single.
+ * \implements std::ranges::View
+ * \implements std::ranges::RandomAccessRange
+ * \implements std::ranges::OutputRange
+ * \ingroup view
+ *
+ * \details
+ *
+ * This class models an infinite range over a given value, although "infinity" is limited by the maximum value of
+ * `difference_type` of which the iterator stores a member variable to represent distances.
+ *
+ * Note that most members of this class are generated by ranges::view_interface which is not yet documented here.
+ */
+template <std::CopyConstructible value_t>
+class repeat_view : public std::ranges::view_interface<repeat_view<value_t>>
+{
+private:
+    //!/brief the base type.
+    using base_t = std::ranges::view_interface<repeat_view<value_t>>;
+
+    //!\brief The sentinel type is set to std::ranges::default_sentinel_t.
+    using sentinel_type = std::ranges::default_sentinel_t;
+
+    //!\brief The iterator type for view::repeat (a random access iterator).
+    template <typename parent_type>
+    class repeat_view_iterator : public detail::random_access_iterator_base<parent_type, repeat_view_iterator>
+    {
+        //!\brief The CRTP base type.
+        using base_t = detail::random_access_iterator_base<parent_type, repeat_view_iterator>;
+
+        //!\brief The base position type.
+        using typename base_t::position_type;
+
+    public:
+        //!\brief Type for distances between iterators.
+        using typename base_t::difference_type;
+        //!\brief Value type of container elements.
+        using typename base_t::value_type;
+        //!\brief Use reference type defined by container.
+        using typename base_t::reference;
+        //!\brief Pointer type is pointer of container element type.
+        using typename base_t::pointer;
+        //!\brief Tag this class as a random access iterator.
+        using typename base_t::iterator_category;
+
+        //!\brief Inherit constructors from base type.
+        using base_t::base_t;
```

Comment:
I am afraid, this only works because the crtp pattern of the ra-iterator-base is wrong.  It should not have public constructors but it has. Can you make a ticket for it? It does not have high priority for the release though. But obviously when we change this, this class will not work as well. So it would be better to have the constructors working anyway. Also, our coding style requires us to also specify all implicitly defined constructors. Note, that the corresponding base class constructors are hidden anyway by the derived class' constructors